src/main/scala/org/youshuren/database/DatabaseImpl.scala

```diff
@@ -0,0 +1,79 @@
+package org.youshuren.database
+
+import doobie._
+import doobie.implicits._
+import cats.effect.IO
+import org.youshuren.config.AppConfig.DBConfig._
+import org.youshuren.model._
+
+trait DatabaseImpl
+
+trait Postgres extends DatabaseImpl {
+
+  import Postgres._
+
+  def insert(user: User): IO[Either[Throwable, String]] = user match {
+    case WeChatUser(id, nickName) =>
+      sql"insert into users values ($id, $nickName)"
```

Comment:
I think it will be more readable by centralizing all SQL queries :)

Reply:
The thing is this sql"..." interpolator provided by Doobie seems can only recognise variables from function parameters. Centralising them will require using other API instead of sql interpolator which may reduce readability.
Referring https://tpolecat.github.io/doobie-0.2.3/15-FAQ.html at "How do I turn an arbitrary SQL string into a Query/Query0?"
Correct me if I'm wrong, I'm rookie in Doobie :P